src/catkin_pkg/package_templates.py

```diff
@@ -106,13 +112,23 @@ def _create_package_template(package_name, description=None, licenses=None,
             if dep.lower() == 'message_generation':
                 if not 'message_runtime' in catkin_deps:
                     sys.stderr.write('WARNING: Packages with messages or services should depend on both message_generation and message_runtime\n')
+                    catkin_deps.append('message_runtime')
```

Comment:
Also changing `catkin_deps` while already iterating over it won't achieve to add `message_runtime` since it won't be iterated over.

Same below.